tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -732,6 +741,11 @@ shared_ptr<CKKSTensor> CKKSTensor::matmul_plain_inplace(
     return shared_from_this();
 }
 
+CKKSTensor CKKSTensor::subscript(const vector<pair<size_t, size_t>>& pairs) {
+    CKKSTensor newTensor = CKKSTensor(this->copy(), pairs);
```

Comment:
Why not running the `_data.subscript` here instead of copying the whole tensor? The constructor could get a TensorStorage ref instead.